sdk/dotnet/Pulumi.Automation/DictionaryContentsComparer.cs

```diff
@@ -0,0 +1,54 @@
+// Copyright 2016-2019, Pulumi Corporation
+
+using System.Collections.Generic;
+
+namespace Pulumi.Automation
+{
+    /// Compares two dictionaries for equality by content, as F# maps would.
+    internal sealed class DictionaryContentsComparer<K, V> : IEqualityComparer<IDictionary<K, V>> where K : notnull
+    {
+        private IEqualityComparer<K> _keyComparer;
+        private IEqualityComparer<V> _valueComparer;
```

Comment:
These can be made `readonly` since they are only assigned once in the constructor

```suggestion
        private readonly IEqualityComparer<K> _keyComparer;
        private readonly IEqualityComparer<V> _valueComparer;
```